src/tools/compiletest/src/runtest.rs

```diff
@@ -3431,11 +3431,24 @@ impl<'test> TestCx<'test> {
         let build_root = self.config.build_base.parent().unwrap().parent().unwrap();
         let build_root = cwd.join(&build_root);
 
-        let tmpdir = cwd.join(self.output_base_name());
-        if tmpdir.exists() {
-            self.aggressive_rm_rf(&tmpdir).unwrap();
+        // We construct the following directory tree for each rmake.rs test:
+        // ```
+        // base_dir/
+        //     rmake.exe
+        //     scratch/
+        // ```
+        // having the executable separate from the scratch directory allows the recipes to
+        // `remove_dir_all(scratch)` without running into permission denied issues because
+        // the executable is not under the `scratch/` directory.
+        //
+        // This setup diverges from legacy Makefile run-make tests.
```

Comment:
Perhaps we can use a simpler name here like "rmake_out"? I'm not sure what "scratch" stands for in this context.

Reply:
Sure, the particular directory name isn't very important here, it just has to not contain `rmake.exe` :3 I will change it to `rmake_out`.